app/controllers/fhir/r4/api_controller.rb

```diff
@@ -457,6 +459,27 @@ def current_client_application
     OauthApplication.find_by(id: doorkeeper_token.application_id) if doorkeeper_token.application_id.present?
   end
 
+  # Determine if the patient's jurisdiction is valid for the requesting application
+  def jurisdiction_valid_for_client?(patient)
+    allowed_jurisdiction_ids = current_client_application.jurisdiction&.subtree&.pluck(:id) ||
+                               current_resource_owner&.jurisdiction&.subtree&.pluck(:id)
+
+    if allowed_jurisdiction_ids.nil?
+      patient.errors.add(:jurisdiction, 'Client application does not have a jurisdiction')
+      false
+    elsif !patient.jurisdiction.nil? && allowed_jurisdiction_ids.include?(patient.jurisdiction[:id])
+      true
+    else
+      patient.errors.add(:jurisdiction, "Jurisdiction must be within the client application's jursdiction")
```

Comment:
I was testing out how this would work with the changes in #495, and I noticed that this is adding the error to `:jurisdiction`, which I'm a little unsure about. As I understand it, `:jurisdiction` is not actually an attribute on a `Patient`, but rather a method that is basically a wrapper for something like: `Jurisdiction.find(jurisdiction_id)`. I've noticed that when I make a `Patient` that doesn't have a `jurisdiction_id` (which is an error because `jurisdiction` is not `optional`), rails automatically places the error on `:jurisdiction_id`, not `:jurisdiction`.

I'm not sure what the best approach is. I will say `:jurisdiction` does seem more natural, but it does kinda break the assumption that I've had that errors will be on actual attributes (or `:base`).

Reply:
Discussed offline, but I've changed it to `jurisdiction_id` for now. It's not ideal, since the value of that property will be meaningless to the client, but at least it won't break things in #495. We'll come up with a better approach, but is good enough for now I think.